SalesforceCommerceCloud/pwa-kit
# @W-14686284@ Fix Preview 403 errors when opening storefront being previewed in a new tab

# Description



**Problem**
When previewing a v3 storefront using Storefront Preview, if the merchant opens the same v3 storefront being previewed in a new browser tab, the storefront opened in the new tab receives 403 error responses for all the SCAPI requests.

This is because the Private SLAS client used in Preview returns a refresh token containing a scope that the Public SLAS client used by the storefront does not have, the SLAS refresh token is configured to be stored in v3 storefronts using cookies. The cookies storage is shared across browser tabs and iframes loading the same domain. Thus both the iframe and the storefront loaded in the new tab use the same SLAS refresh token.

STR:
1. Open Storefront Preview in Runtime Admin to preview a v3 storefront. 
2. Open that same v3 storefront in a new browser tab.
3. Observe the 403 error responses for all the SCAPI requests in the storefront opened in a new tab.

**Workaround**
The workaround is to use a different cookie key `cc-nx-g-iframe` and `cc-nx-iframe` to store the SLAS refresh token when previewing the storefront. Using different cookie keys separates browser contexts, meaning that different sessions can be active in the browser tabs and the iframe, even if they are loading the same domain.



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Store SLAS refresh token in a different cookie key when previewing the storefront to keep separated browser contexts.

# How to Test-Drive This PR

1. Open Storefront Preview in Runtime Admin to preview a v3 storefront with the fix. 
2. Open that same v3 storefront with the fix in a new browser tab.
3. Observe there are no 403 error responses for the SCAPI requests in the storefront opened in a new tab, and the context is separated, i.e. the context set in Preview is not reflected in the storefront opened in a new tab.

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
